### ISSUE TITLE
Could not find user: admin, code\": 401, \"title\": \"Unauthorized

### ISSUE THREAD

**Issue by phijojoseph** [fog/fog#3699]
i tried to use identity v3 api, but it is failing for me and throwing some error stated above.
Not sure that this is  a bug or problem with my code.Request for  help. If someone has got a successful response on v3 auth token please post the code here.
I am using openstack  kilo.
Please find my code and output below
 **************Code**************
require 'fog'
require 'pp'
auth_url= 'http://127.0.0.1:15000/v3/auth/tokens'
username = 'admin'
password = 'root123'
keystone = Fog::Identity.new({
    :provider            => 'Openstack',    
    #:openstack_management_url =>  auth_url,                                 # OpenStack Fog provider
    :openstack_auth_url  => auth_url, # OpenStack Keystone endpoint
    :openstack_username  => username,                                  # Your OpenStack Username
    :openstack_api_key   => password,                              # Your OpenStack Password
    :openstack_domain_id => 'default',
    #:connection_options  => {}                                                # Optional
})

**********Output *******

ruby openstack_services.rb
/Users/phijo/.rvm/gems/ruby-2.0.0-p353/gems/excon-0.45.4/lib/excon/middlewares/expects.rb:6:in `response_call': Expected([201]) <=> Actual(401 Unauthorized) (Excon::Errors::Unauthorized)
excon.error.response
  :body          => "{\"error\": {\"message\": \"Could not find user: admin (Disable debug mode to suppress these details.)\", \"code\": 401, \"title\": \"Unauthorized\"}}"
  :headers       => {
    "Content-Length"         => "138"
    "Content-Type"           => "application/json"
    "Date"                   => "Tue, 22 Sep 2015 15:39:32 GMT"
    "Server"                 => "Apache/2.4.6 (CentOS) OpenSSL/1.0.1e-fips mod_wsgi/3.4 Python/2.7.5"
    "Vary"                   => "X-Auth-Token"
    "WWW-Authenticate"       => "Keystone uri=\"http://127.0.0.1:15000\""
    "x-openstack-request-id" => "req-01995e0f-1beb-4b72-a91f-180f351f3698"
  }
  :local_address => "127.0.0.1"
  :local_port    => 51266
  :reason_phrase => "Unauthorized"
  :remote_ip     => "127.0.0.1"
  :status        => 401
  :status_line   => "HTTP/1.1 401 Unauthorized\r\n"
	from /Users/phijo/.rvm/gems/ruby-2.0.0-p353/gems/excon-0.45.4/lib/excon/middlewares/response_parser.rb:8:in `response_call'
	from /Users/phijo/.rvm/gems/ruby-2.0.0-p353/gems/excon-0.45.4/lib/excon/connection.rb:372:in `response'
	from /Users/phijo/.rvm/gems/ruby-2.0.0-p353/gems/excon-0.45.4/lib/excon/connection.rb:236:in `request'
	from /Users/phijo/.rvm/gems/ruby-2.0.0-p353/gems/fog-core-1.32.1/lib/fog/core/connection.rb:81:in `request'
	from /Users/phijo/.rvm/gems/ruby-2.0.0-p353/gems/fog-1.34.0/lib/fog/openstack/core.rb:509:in `retrieve_tokens_v3'
	from /Users/phijo/.rvm/gems/ruby-2.0.0-p353/gems/fog-1.34.0/lib/fog/openstack/core.rb:298:in `authenticate_v3'
	from /Users/phijo/.rvm/gems/ruby-2.0.0-p353/gems/fog-1.34.0/lib/fog/openstack/core.rb:175:in `authenticate'
	from /Users/phijo/.rvm/gems/ruby-2.0.0-p353/gems/fog-1.34.0/lib/fog/openstack/core.rb:136:in `authenticate'
	from /Users/phijo/.rvm/gems/ruby-2.0.0-p353/gems/fog-1.34.0/lib/fog/openstack/identity_v3.rb:163:in `initialize'
	from /Users/phijo/.rvm/gems/ruby-2.0.0-p353/gems/fog-core-1.32.1/lib/fog/core/service.rb:115:in `new'
	from /Users/phijo/.rvm/gems/ruby-2.0.0-p353/gems/fog-core-1.32.1/lib/fog/core/service.rb:115:in `new'
	from /Users/phijo/.rvm/gems/ruby-2.0.0-p353/gems/fog-1.34.0/lib/fog/openstack/identity.rb:14:in `new'
	from /Users/phijo/.rvm/gems/ruby-2.0.0-p353/gems/fog-core-1.32.1/lib/fog/core/services_mixin.rb:16:in `new'
	from openstack_services.rb:7:in `<main>'

Expecting an answer
Regards
Phijo

**Comment by geemus** [comment]
@dhague @TerryHowe any advice?

**Comment by dhague** [comment]
I ran the same code against my local Kilo devstack instance and it worked OK.

The clue here is in the error message:
>Could not find user: admin, code\": 401, \"title\": \"Unauthorized

This indicates that your OpenStack instance does not have the user "admin" in the domain "default".

You may be able to fix this with the following command on your Kilo system:
>openstack role add --user admin --domain default admin

Best regards,
Darren

P.S. Signing off with "Expecting an answer" could be considered rude in some cultures, especially when addressed to a community providing software for free. "Hoping for an answer" might be a better choice of words.

**Comment by phijojoseph** [comment]
@dhague First of all sorry for the above statement. my language is not good , Please don't consider it as an rude signing off. Actually i didn't really mean that.  

Thank you for guiding me . but above command is not working for me  --domain is not recognised 

>  openstack role add: error: unrecognized arguments: --domain admin

**Comment by dhague** [comment]
OK, I think I see what's going on.

The Fog code generates this request:
>{
>	"auth" : {
>		"identity" : {
>			"methods" : ["password"],
>			"password" : {
>				"user" : {
>					"password" : "password",
>					"domain" : {
>						"id" : "default"
>					},
>					"name" : "admin"
>				}
>			}
>		},
>		"scope" : {
>			"domain" : {
>				"id" : "default"
>			}
>		}
>	}
>}

In other words, Fog adds the `scope` element at the end, scoping the token to the domain. This is because the parameter `:openstack_domain_id` is set.

If I change this parameter to `:openstack_user_domain_id` then only the user is scoped to the domain (not the token) and then I get exactly the same JSON request as you show above.

So, in summary: when you call Fog::Identity.new() just change the parameter `:openstack_domain_id` to `:openstack_user_domain_id` to solve the problem.

Hope this helps you,
Darren

**Comment by phijojoseph** [comment]
@dhague It works, thank you for the support :100: :+1:
